### PR TITLE
Initial commit of package version check before testing

### PR DESCRIPTION
**Pull request by rocktavious**
## Issues

Add the PackageVersion check to the library

## Changelog

- [X] List your changes here
- [X] Make a `changie` entry

### CODE REVIEW

File: check_package_version.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package opslevel
+
+type PackageVersionCheckFragment struct {
+	MissingPackageResult       *CheckResultStatusEnum `graphql:"missingPackageResult"`       // The check result if the package isn't being used by a service.
+	PackageConstraint          *PackageConstraintEnum `graphql:"packageConstraint"`          // The package constraint the service is to be checked for.
```
Comment (davidbloss) [inline check_package_version.go:5]: Should `PackageConstraint` be a pointer? Looks like it's non-nullable in the docs

Comment (rocktavious) [inline check_package_version.go:5]:
```suggestion
	PackageConstraint          PackageConstraintEnum `graphql:"packageConstraint"`          // The package constraint the service is to be checked for.
```

File: check_package_version.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	PackageManager             PackageManagerEnum     `graphql:"packageManager"`             // The package manager (ecosystem) this package relates to.
+	PackageName                string                 `graphql:"packageName"`                // The name of the package to be checked.
+	PackageNameIsRegex         bool                   `graphql:"packageNameIsRegex"`         // Whether or not the value in the package name field is a regular expression.
+	VersionConstraintPredicate Predicate              `graphql:"versionConstraintPredicate"` // The predicate that describes the version constraint the package must satisfy.
```
Comment (rocktavious) [inline check_package_version.go:9]:
```suggestion
	VersionConstraintPredicate *Predicate              `graphql:"versionConstraintPredicate"` // The predicate that describes the version constraint the package must satisfy.
```

**Review by davidbloss — APPROVED** [review]
LGTM 🎉

**Pull request merged**